up-streamer/src/ustreamer.rs

```diff
@@ -0,0 +1,973 @@
+
+                    if let Err(err) = unregister_res.as_ref() {
+                        error!(
+                            "{}:{}:{} Deleting forwarding rule failed: {:?}",
+                            self.name, USTREAMER_TAG, USTREAMER_FN_DELETE_FORWARDING_RULE_TAG, err
+                        );
+                    } else {
+                        debug!(
+                            "{}:{}:{} Deleting forwarding rule succeeded",
+                            self.name, USTREAMER_TAG, USTREAMER_FN_DELETE_FORWARDING_RULE_TAG
+                        );
+                    }
+                    unregister_res
+                } else {
+                    let no_in_transport_out_authority_found = UStatus::fail_with_code(
+                        UCode::NOT_FOUND,
+                        "No in_transport and out_authority found corresponding",
+                    );
+                    error!(
+                        "{}:{}:{} Deleting forwarding rule failed: {:?}",
+                        self.name,
+                        USTREAMER_TAG,
+                        USTREAMER_FN_DELETE_FORWARDING_RULE_TAG,
+                        &no_in_transport_out_authority_found
+                    );
+                    Err(no_in_transport_out_authority_found)
+                }
+            } else {
+                Ok(())
+            }
+        } else {
+            let err = Err(UStatus::fail_with_code(
+                UCode::NOT_FOUND,
+                format!(
+                    "{} is not routed. No rule to delete.",
+                    Self::forwarding_id(&r#in, &out)
+                ),
+            ));
+            error!(
+                "{}:{}:{} Deleting forwarding rule failed: {:?}",
+                self.name, USTREAMER_TAG, USTREAMER_FN_DELETE_FORWARDING_RULE_TAG, err
+            );
+            err
+        }
+    }
+}
+
+#[derive(Clone)]
+pub(crate) struct ComparableTransport {
+    transport: Arc<Box<dyn UTransport>>,
+}
+
+impl ComparableTransport {
+    pub fn new(transport: Arc<Box<dyn UTransport>>) -> Self {
+        Self { transport }
+    }
+}
+
+impl Hash for ComparableTransport {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        Arc::as_ptr(&self.transport).hash(state);
+    }
+}
+
+impl PartialEq for ComparableTransport {
+    fn eq(&self, other: &Self) -> bool {
+        Arc::ptr_eq(&self.transport, &other.transport)
+    }
+}
+
+impl Eq for ComparableTransport {}
+
+const TRANSPORT_FORWARDER_TAG: &str = "TransportForwarder:";
+const TRANSPORT_FORWARDER_FN_MESSAGE_FORWARDING_LOOP_TAG: &str = "message_forwarding_loop():";
+pub(crate) struct TransportForwarder {}
+
+impl TransportForwarder {
+    async fn new(
+        out_transport: Arc<Box<dyn UTransport>>,
+        message_receiver: Receiver<Arc<UMessage>>,
+    ) -> Self {
+        let out_transport_clone = out_transport.clone();
+        let message_receiver_clone = message_receiver.clone();
+        task::spawn(Self::message_forwarding_loop(
```

Comment:
@sophokles73 -- I was reading last week that for a long-running (forever) task such as this one, it's a bad fit for spawning a new task like this because there's only a limited number of threads in that thread-pool.

I read it's better to spawn a dedicated thread so that we're not "stealing" one from the thread pools the async executor starts in [this nice article](https://ryhl.io/blog/async-what-is-blocking/#:~:text=iterators%20are%20blocking.-,Spawn%20a%20dedicated%20thread,-If%20a%20blocking).

So I think I need to update and fix this.

WDYT?

Reply:
Went and updated this.